.devcontainer/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Note: You can use any Debian/Ubuntu based image you want. 
+FROM mcr.microsoft.com/vscode/devcontainers/base:0-bullseye
+
+# [Option] Install zsh
+ARG INSTALL_ZSH="true"
+# [Option] Upgrade OS packages to their latest versions
+ARG UPGRADE_PACKAGES="false"
+# [Option] Enable non-root Docker access in container
+ARG ENABLE_NONROOT_DOCKER="true"
+# [Option] Use the OSS Moby CLI instead of the licensed Docker CLI
+ARG USE_MOBY="true"
+# [Option] Select CLI version
+ARG CLI_VERSION="latest"
+
+# Enable new "BUILDKIT" mode for Docker CLI
+ENV DOCKER_BUILDKIT=1
+
+# Install needed packages and setup non-root user. Use a separate RUN statement to add your
+# own dependencies. A user of "automatic" attempts to reuse an user ID if one already exists.
+ARG USERNAME=automatic
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+COPY library-scripts/*.sh /tmp/library-scripts/
+RUN apt-get update \
+    && /bin/bash /tmp/library-scripts/common-debian.sh "${INSTALL_ZSH}" "${USERNAME}" "${USER_UID}" "${USER_GID}" "${UPGRADE_PACKAGES}" "true" "true" \
```

Comment:
great work. will it be better to use curl / wget to [download and run](https://github.com/microsoft/vscode-dev-containers/tree/main/script-library#downloading-the-script-with-curl--wget-instead) those scripts instead of checking them in to the repo?

Reply:
Or can we use this container directly? `mcr.microsoft.com/vscode/devcontainers/javascript-node`

Reply:
@gao-sun ye, it's simple to set this and rebuild container by Codespaces' plugin, and according to @wangsijie demand, this PR don't need to merge to project, a section of WIKI is fine enough

So, I'll close this PR, and it's fine to reopen if you guys need this

Reply:
@bxb100 I think it's still worthy to have a container config if it'll be out-of-box for developers, like what we did for gitpod.

Reply:
I don't know much about codespaces, feel free to correct me if my comment was wrong.